packages/publisher/bitbucket/src/PublisherBitbucket.ts

```diff
@@ -0,0 +1,66 @@
+import PublisherBase, { PublisherOptions } from '@electron-forge/publisher-base';
+import { asyncOra } from '@electron-forge/async-ora';
+
+import fetch from 'node-fetch';
+import FormData from 'form-data';
+import fs from 'fs-extra';
+
+import { PublisherBitbucketConfig } from './Config';
+
+export default class PublisherGithub extends PublisherBase<PublisherBitbucketConfig> {
+  name = 'bitbucket';
+
+  async publish({ makeResults }: PublisherOptions) {
+    const { config } = this;
+    const hasRepositoryConfig = config.repository && typeof config.repository;
+    const hasAuthConfig = config.auth && typeof config.auth === 'object';
+    let appPassword = process.env.BITBUCKET_APP_PASSWORD;
+    let bbUsername = process.env.BITBUCKET_USERNAME;
+
+    if (!(hasRepositoryConfig && config.repository.owner && config.repository.name)) {
+      throw 'In order to publish to Bitbucket you must set the "repository.owner" and "repository.name" properties in your forge config. See the docs for more info'; // eslint-disable-line
+    }
+
+    if (!(config.auth && typeof config.auth === 'object') && !appPassword && !bbUsername) {
+      throw 'In order to publish to Bitbucket you must set the "auth" object in your forge config or use BITBUCKET_APP_PASSWORD and BITBUCKET_USERNAME environment variables';
+    }
+
+    appPassword = config.auth.appPassword || process.env.BITBUCKET_APP_PASSWORD;
+    bbUsername = config.auth.username || process.env.BITBUCKET_USERNAME;
+
+    if (!appPassword || !bbUsername) {
+      throw 'In order to publish to Bitbucket you must set the "auth.appPassword" and "auth.username" properties in your forge config.';
+    }
+
+    for (const [index, makeResult] of makeResults.entries()) {
+      const data = new FormData();
+
+      let i = 0;
+      for (const artifactPath of makeResult.artifacts) {
+        data.append('files', fs.createReadStream(artifactPath));
+        i += 1;
+      }
+
+      // TODO: Consider checking if the files already exist at the current version and abort if so?
```

Comment:
Maybe add an `overwrite` option to the config, if `true` just upload, if `false` check if it exists before uploading.

Reply:
Sweet, want that now? Config the place to do it? Or is there a way for a specific publisher to take a flag? Probably config is more consistent?

Reply:
Deciding between throwing or prompting in the other case.
I'm leaning towards throwing, because it's likely we'd want to call this in CI where you wont be able to prompt anyway

Reply:
Done